src/batch.rs

```diff
@@ -27,61 +24,130 @@ fn gen_128_bits<R: RngCore + CryptoRng>(mut rng: R) -> [u64; 4] {
     bytes
 }
 
+enum Inner {
+    SpendAuth {
+        vk_bytes: VerificationKeyBytes<SpendAuth>,
+        sig: Signature<SpendAuth>,
+        c: Scalar,
+    },
+    Binding {
+        vk_bytes: VerificationKeyBytes<Binding>,
+        sig: Signature<Binding>,
+        c: Scalar,
+    },
+}
+
+impl
+    From<(
+        VerificationKeyBytes<SpendAuth>,
+        Signature<SpendAuth>,
+        Scalar,
+    )> for Inner
+{
+    fn from(
+        tup: (
+            VerificationKeyBytes<SpendAuth>,
+            Signature<SpendAuth>,
+            Scalar,
+        ),
+    ) -> Self {
+        let (vk_bytes, sig, c) = tup;
+        Inner::SpendAuth { vk_bytes, sig, c }
+    }
+}
+
+impl From<(VerificationKeyBytes<Binding>, Signature<Binding>, Scalar)> for Inner {
+    fn from(tup: (VerificationKeyBytes<Binding>, Signature<Binding>, Scalar)) -> Self {
+        let (vk_bytes, sig, c) = tup;
+        Inner::Binding { vk_bytes, sig, c }
+    }
+}
+
```

Comment:
Where do these impls get used? Below, it looks like the `impl From<_> for Item` impls construct `Inner` instances directly -- could these ones be eliminated, or am I missing something?

Reply:
I think these were conveniences for my submethod so they are now defunct, removing.